[문자열 덧셈 계산기] 조재중 미션 제출합니다.

m-a-king opened the pull request:
초심으로 돌아가서 **과하지도 않고**, **너무 미흡하지도 않은** 코드를 작성하기 위해 노력했습니다. 
즉, 코드의 **가독성**과 **유지보수성**을 유지하면서도, **불필요한 복잡성**을 피하고자 했습니다.

감사합니다.

m-a-king committed:
docs(README.md): add 기능요구사항, 입력형식제약, 입출력요구사항, 예외처리

기능 요구사항, 입력 형식 제약, 입출력 예시, 예외 처리 섹션을 README.md 파일에 추가했습니다.
특히 입력 형식 제약을 통해 명시되지 않은 엣지 케이스들을 처리하는 방법을 구체적으로 정의했습니다.
이를 통해 입력 유효성 검사를 강화하고, 예상치 못한 입력에 대한 처리를 명확하게 할 수 있습니다.

m-a-king committed:
docs(README.md): update 기능요구사항, 입력형식제약, 입출력요구사항

커스텀 구분자 활성화 시 기본 구분자가 비활성화됨을 명시.
기능 요구사항, 입력 형식 제약, 입출력 예시 섹션을 더 명확히 개선.
실행 결과 예시를 가독성 개선.

m-a-king committed:
docs(README.md): add 프로그래밍 요구사항 섹션

프로그래밍 요구사항에 JDK 버전, 프로그램 시작점, 외부 라이브러리 제한, 코드 컨벤션 등의 내용을 추가함.

m-a-king committed:
docs(README.md): update 기능 요구 사항

예시에 BigDecimal 처리를 추가.

m-a-king committed:
docs(README.md): add 프로그램 전체 흐름, 메서드 역할과 예외 처리, 메서드 호출 흐름

프로그램의 전체적인 처리 흐름을 설명.
각 메서드의 역할과 예외 처리 방식을 설명.
메서드 호출 순서와 흐름을 설명.

m-a-king committed:
docs(README.md): add Activity Diagram

프로그램의 처리 흐름을 시각적으로 나타내는 활동 다이어그램을 추가하여 각 단계와 예외 처리 과정을 쉽게 이해할 수 있도록 함.

m-a-king committed:
docs(README.md): update Activity Diagram

불필요한 소괄호 제거

m-a-king committed:
docs(README.md): delete Activity Diagram

docs(README.md): delete Activity Diagram

절차지향적으로 작성된 액티비티 다이어그램이 현재 코드의 흐름과 부적합함을 인지하고 삭제함.

m-a-king committed:
docs(README.md): refactor 클래스 설계(역할과 책임 기반)

기존의 절차지향적인 메서드 역할과 호출 흐름을 개선하여 역할과 책임을 분리한 클래스 설계로 변경.
InputOutputHandler, DelimiterManager, NumberProcessor 등 각 클래스를 도입해
입력/출력, 구분자 처리, 숫자 처리의 역할을 명확히 구분.

m-a-king committed:
feat(IOService): add 입출력 처리 기능

입력을 받아 유효성을 검사하고, 결과를 출력하는 IOService 클래스 추가
- 입력 값에 공백이 포함되면 예외 처리
- 계산 결과 출력 메서드 추가

m-a-king committed:
feat(DelimiterService): add 구분자 처리 및 숫자 추출 기능

입력 문자열을 처리하여 기본 구분자(쉼표, 콜론) 또는 커스텀 구분자를 기준으로 숫자를 추출하는 DelimiterService 클래스 추가
- 커스텀 구분자 유효성 검사 및 정규식 패턴 반환.
- 기본 구분자 쉼표와 콜론을 상수로 설정하고 이를 정규식 패턴으로 반환.
- 입력 문자열을 처리하여 구분자를 기준으로 숫자를 추출.

m-a-king committed:
feat(NumberService): add 숫자 처리 및 합계 계산 기능

숫자 리스트를 처리하고 유효성을 검사하며, 합계를 계산하는 NumberService 클래스 추가
- 모든 양수를 처리하기 위해 BigDecimal 사용.
- 빈 값이나 불필요한 0이 포함된 숫자 예외 처리.
- 음수 및 0을 허용하지 않음.
- 숫자의 합계를 계산하는 메서드 추가.

m-a-king committed:
feat(Coordinator): add 전체 프로그램 실행 조정 기능

IOService, DelimiterService, NumberService의 기능을 조정하여 전체 프로그램 실행을 관리하는 Coordinator 클래스 추가
- 입력을 받아 구분자 처리 및 숫자 추출
- 숫자 변환 및 합산 처리
- 결과 출력
- 잘못된 입력이나 오류 발생 시 예외 처리 로직 추가

m-a-king committed:
feat(Application): add 메인 클래스 및 프로그램 실행 흐름

프로그램 실행을 위한 Application 클래스
- IOService, DelimiterService, NumberService의 인스턴스를 생성하고, Coordinator에 주입하여 실행
- 입력 -> 구분자 처리 -> 숫자 변환 및 계산 -> 결과 출력 흐름을 담당

m-a-king committed:
docs(README.md): change 구분자 위치 제약

기존의 구분자가 문자열의 처음이나 끝에 위치할 수 없다는 제약을 제거하고, 이를 허용하도록 입력 형식 제약 변경

m-a-king committed:
style(IOService): 불필요하거나 잘못된 주석 제거

입력 유효성 검사 메서드에서 주석이 실제 코드와 맞지 않아 이를 제거함.

m-a-king committed:
docs(README.md): change 구분자 연속 사용 제약

구분자가 연속적으로 사용될 수 없다는 제약을 제거하고, 이를 허용하도록 입력 형식 제약 변경

m-a-king committed:
feat(IOService): update 소수 출력 형식

변경 전: 결과: 1E-9
변경 후: 결과: 0.000000001

m-a-king committed:
feat(NumberService): change 공백 처리 예외

구분자 사이에 값이 없거나, 구분자와 숫자 모두 없이 공백이 입력되는 케이스를 처리하기 위해서 예외 처리 변경

m-a-king committed:
feat(IOService): update 출력 메시지에 공백 추가

변경 전: '결과: 6'
변경 후: '결과 : 6'

m-a-king committed:
feat(Coordinator): add 예외 처리 후 예외 다시 던지기

잘못된 입력 또는 오류 발생 시 로그를 출력한 후 Application(main)으로 예외를 다시 던지도록 수정함.

m-a-king committed:
test(InputTest): add 다양한 입력 및 예외 처리 테스트 케이스

- 기본 구분자 및 커스텀 구분자 사용 시 정상 동작 테스트 추가
- 커스텀 구분자 사용 (세미콜론, 문자 'a')에 대한 테스트 추가
- 잘못된 입력 (구분자가 여러 개인 경우, 구분자가 숫자인 경우, 공백 포함)에 대한 예외 처리 테스트 추가
- 음수 입력에 대한 예외 처리 테스트 추가

m-a-king committed:
docs(README): update 클래스 접미사

변경 전: Handler, Manager, Processor
변경 후: Service

m-a-king committed:
docs(README): update 클래스 접미사

변경 전: Handler, Manager, Processor
변경 후: Service

ehgur062300 commented on src/main/java/calculator/IOService.java:
```diff
@@ -0,0 +1,25 @@
+        return input;
+    }
+
+    private void validateInput(String input) {
```
"" 제가 알기로는 Console.readLine()에 입력값이 없으면 NoSuchElementException 이 발생하는 걸로 알고있어요!
따로 처리를 하셨는지 궁금합니다!

m-a-king replied:
방금 실행해보니 정상작동되네요.

ehgur062300 commented on src/main/java/calculator/NumberService.java:
```diff
@@ -0,0 +1,37 @@
+    public List<BigDecimal> processNumbers(List<String> numberStrings) {
+        return numberStrings.stream()
+                .map(numString -> {
+                    try {
```
우테코에서 자바 컨벤션을 지켜달라는 얘기 중 클린코드 내용이 있습니다
하나의 메서드에 depth를 3이상 가져가지 말라는 말이있는데 한번 참고해보셔도 좋을거 같아요!
https://github.com/woowacourse/woowacourse-docs/blob/main/cleancode/pr_checklist.md

m-a-king replied:
감사합니다. 인지하고 있었는데 stream에서 문제가 생겼네요. 코드 냄새.. 확인했습니다.

ehgur062300 reviewed (commented):
1주차 고생하셨습니다.!!!

m-a-king committed:
feat(IOService): add Console(Scanner) 자원 해제 처리

입력 후 Console(Scanner) 자원이 제대로 닫히도록 finally 블록에 close 동작을 추가.

m-a-king committed:
refactor(NumberService): 숫자 변환과 유효성 검사 로직 분리

숫자 문자열을 BigDecimal로 변환하는 로직과 유효성 검사 로직을 분리.
코드의 가독성과 유지보수성을 개선.

m-a-king committed:
feat(NumberService): add 빈 문자열을 BigDecimal(0)으로 처리

빈 문자열이 입력될 경우, BigDecimal.ZERO를 반환하도록 처리하여 빈 문자열에 대한 예외를 방지.

m-a-king committed:
test(InputTest): refactor 테스트 메서드 네이밍 개선

테스트 메서드 이름을 더 직관적으로 수정하여 가독성 개선.

m-a-king committed:
test(InputTest): add 빈 문자열 입력

빈 문자열 입력 시 0을 반환하는 테스트 추가.

m-a-king committed:
feat(IOService): add 빈 문자열 입력 처리

빈 입력 시 발생할 수 있는 NoSuchElementException을 처리하여 안전하게 빈 문자열로 변환.

m-a-king committed:
docs(README): add 회고록 링크, 우테코 이미지

m-a-king committed:
docs(README): add 회고록 링크, 우테코 이미지

m-a-king committed:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	README.md

NaMinhyeok commented on src/main/java/calculator/Application.java:
```diff
@@ -2,6 +2,19 @@
 public class Application {
     public static void main(String[] args) {
-        // TODO: 프로그램 구현
+        // 입력과 출력 담당
+        IOService ioService = new IOService();
+
+        // 구분자 처리 담당
+        DelimiterService delimiterService = new DelimiterService();
+
+        // 숫자 처리 및 계산 담당
+        NumberService numberService = new NumberService();
+
+        // Coordinator에 필요한 의존성 주입
+        Coordinator coordinator = new Coordinator(ioService, delimiterService, numberService);
+
+        // 프로그램 실행
+        coordinator.execute();
```
주석이라는게 사실 굉장히 좋은 의미를 가지지만 주석은 양면성을 가질 수 있기 때문에 충분히 new와 각 클래스들의 이름으로 무슨 역할을 하는지 파악 할 수 있으니 지우셔도 좋을 것 같아요 !

주석으로 표현하는 것도 좋지만 최대한 주석의 내용을 코드로 풀어내는게 중요하다고 생각합니다 !

m-a-king replied:
감사합니다. 참고하겠습니다 ㅎㅎ

NaMinhyeok commented on src/main/java/calculator/Coordinator.java:
```diff
@@ -0,0 +1,39 @@
+            // 1. 입력 받기
+            String input = IOService.getInput();
+
+            // 2. 구분자 처리 및 숫자 추출
+            List<String> numberStrings = delimiterService.processInputAndExtractNumbers(input);
+
+            // 3. 숫자 처리 및 계산
+            List<BigDecimal> numbers = numberService.processNumbers(numberStrings);
+            BigDecimal result = numberService.calculateSum(numbers);
```
이거는 제 개인적인 생각인데 만약 더 추상화한다면 주석이 이렇게 묶일게 아니라

```suggestion
            // 2. 구분자 처리 및 숫자 추출
            List<String> numberStrings = delimiterService.processInputAndExtractNumbers(input);
            List<BigDecimal> numbers = numberService.processNumbers(numberStrings);
            
            // 3. 숫자 계산
            BigDecimal result = numberService.calculateSum(numbers);
```

이게 좀 더 좋지않을까요 ? 그냥 코드스타일의 차이인것 같긴합니다 ㅎㅎ

m-a-king replied:
그렇네요.. 코드 수정하면서 주석을 알맞게 수정하지 못한 것, 확인해주셔서 감사합니다.

NaMinhyeok commented on src/main/java/calculator/DelimiterService.java:
```diff
@@ -0,0 +1,49 @@
+    // 커스텀 구분자를 추출하고 정규식 패턴으로 반환하는 메서드
+    private String getCustomDelimiterPattern(String customDelimiterLine) {
+        // 커스텀 구분자 형식 체크
+        if (!customDelimiterLine.startsWith("//") || customDelimiterLine.length() != 3) {
+            throw new IllegalArgumentException("잘못된 구분자 형식입니다.");
+        }
```
```suggestion
        if (isCustomDelimiter()) {
            throw new IllegalArgumentException("잘못된 구분자 형식입니다.");
        }
```
이런식으로 추상화하면 더 좋을 것 같아요 ! 그리고 매직넘버나 매직스트링을 상수처리하면 좋을 것 같아요 ! "//" 같은 경우는 상수로 관리한다면 추후 유지보수에도 좋다고 생각합니다 !

m-a-king replied:
네 민혁님 코드 보면서 느꼈습니다. 매직넘버, 스트링 관리를 조금 더 신경 써야 할 것 같아요.

NaMinhyeok commented on src/main/java/calculator/IOService.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    private void validateInput(String input) {
+        if (input.contains(" ")) {
+            throw new IllegalArgumentException("공백을 입력할 수 없습니다.");
+        }
```
제가 알기로는 input.contains를 하는 것보다는 input.isBlank() 를 쓰는게 더 좋을 것 같아요 ! 이게 충분히 공백이 존재하는지 확인해줄 수 있어요 !

혹시라도 다른 곳에서 예를 들어 1,   2:   3 을 못하게 하는거라면 이런 방식도 괜찮은 것 같아요 !

m-a-king replied:
후자가 맞습니다! 제가 임의로 설정해뒀고, 공백 입력 자체를 금지하는 것입니다.

NaMinhyeok commented on src/main/java/calculator/NumberService.java:
```diff
@@ -0,0 +1,49 @@
+
+        BigDecimal number = new BigDecimal(numString);
+
+        if (number.compareTo(BigDecimal.ZERO) <= 0) {
+            throw new IllegalArgumentException("음수나 0은 허용되지 않습니다.");
+        }
```
아 이걸 보니까 0이 허용이 안되는군요.. 생각해보니까 양수라고했으니 0은 뺐어야되는데 이걸 보고 캐치했네요 ㅠㅠ

m-a-king replied:
다행이네요 ㅎㅎ

NaMinhyeok commented on src/main/java/calculator/NumberService.java:
```diff
@@ -0,0 +1,49 @@
+
+public class NumberService {
+
+    public List<BigDecimal> processNumbers(List<String> numberStrings) {
```
저는 BigDecimal을 쓰는것도 나쁘다고는 생각안하는데 BigDecimal을 쓰는 이유가 있었으면 좋겠다고 생각합니다 ! 
저도 요구사항을 모두 지키지 못한 것 같긴한데... BigDecimald은 부동소수점문제를 해결하기 위해서 고정소수점 방식을 사용하기 위해서 사용하는 것인데 현재 요구사항에서는 양수합산만을 요구하고있습니다 ! 소수가 나올 일이 없기 때문에 굳이 BigDecimal이어야 했냐? 라고 한다면 그에 답변 할 수 있어야 한다고 해요 !

이거에 대한 답을 하기 위해서는 BigDecimal을 사용하돼 VO 처럼 사용해서 그곳에서 양수인지 검증해주고 있다면 다음에 유지보수를 위해서 소수를 사용해야 하는 순간에 양수인지 검증하는 메서드만 변경해주면 되지않을까? 라고 생각합니다 ! 

추후 다른 로직이 추가될 걸 생각해서 BigDecimal을 썼다면 그에 맞게 이해되는 로직이 더 추가되었어야된다고 생각합니다 ! 지금 로직상으로는 문제는 없어보여요 ! 
그래도 고민하고 쓰셨다는게 대단하십니다... 저는 오히려 BigDecimal을 보고서야 이러한 생각이 들었거든요 ㅎㅎ

m-a-king replied:
저는 양수에 1111111111111111111111111111111111.1 과 같이 큰 값이 들어와도 처리해야한다라고 파악해서 BigDecimal 사용했습니다.

그리고 양수에 소수도 포함이 되어서 BigInteger가 아닌 BigDecimal 사용했습니다.

NaMinhyeok replied:
양수에 소수가 포함이되나요 ㅠㅠ 수학적 지식의 밑천이 드러났네요 ㅎㅎ

조금 부끄러운 리뷰가 되었는데 하고자하는 말은 내가 작성한 코드에 이유가 있어야한다는 것이었습니다! 우테코에서도 얘기하던 메타인지처럼 코드를 왜? 라고 의심하면서 작성하자는 것이죠

부끄럽네요 ㅎㅎ 알고 보니까 잘 작성된 코드가 된 것 같습니다

NaMinhyeok reviewed (commented):
되게 문서화를 잘하신 것 같아서 부럽네요.. 저도 이렇게 README를 먼저 작성해두고 하고싶었는데 잘 안되더라구요 대체로 작성을 잘 하신것 같아서 몇가지 사항만 달아드렸습니다 !
전체적으로 매직넘버나 매직스트링을 상수화하시고, 메서드들을 좀 더 추상화해보는게 어떨까 생각합니다!

1주차 고생하셨습니다~ 2주차 화이팅하세용

NaMinhyeok commented on src/test/java/calculator/InputTest.java:
```diff
@@ -0,0 +1,88 @@
+
+class InputTest extends NsTest {
+
+    // 빈 문자열 입력 처리
```
아이코 이건 깜빡했네요 이렇게 주석쓰시기 보다는 DisplayName을 쓰시는게 더 좋지않을까 생각합니다 ㅎㅎ

m-a-king replied:
네 테스트 코드 작성, 조금 더 연습이 필요하고 예제를 많이 참고해야할 것 같습니다. 감사합니다.

m-a-king committed:
refactor(Application): 메인 메서드 간소화 및 주석 제거

- 클래스 이름만으로 각자의 역할이 명확하므로 불필요한 주석을 제거.
- 의존성 주입 부분을 한 줄로 간소화하여 코드 가독성 향상.

m-a-king committed:
chore(Coordinator): 메서드 호출 위치 개선

- processNumbers 메서드가 올바른 주석 아래에서 호출되도록 수정

m-a-king committed:
refactor(DelimiterService): 커스텀 구분자 검증 조건을 추출

- `isValidCustomDelimiter` 메서드를 추가하여 커스텀 구분자 검증 로직을 긍정형으로 변경
- 가독성을 높이고 명확한 로직 표현을 위해 코드 구조 개선

m-a-king committed:
test(InputTest): add @DisplayName 애너테이션, 테스트 설명 개선

- 각 테스트 메서드에 @DisplayName 애너테이션을 적용하여 테스트의 목적을 명확히 표현
- 테스트 실행 시 가독성을 향상

yeonjookang commented on src/main/java/calculator/Coordinator.java:
```diff
@@ -0,0 +1,39 @@
+            // 4. 결과 출력
+            IOService.printResult(result);
+        } catch (IllegalArgumentException e) {
+            System.err.println("잘못된 입력: " + e.getMessage());
```
출력부분은 IOService의 역할인 것 같아요!

m-a-king replied:
오... 맞아요. 완전히 놓쳤네요! 감사합니다.

yeonjookang commented on src/main/java/calculator/DelimiterService.java:
```diff
@@ -0,0 +1,53 @@
+    private String getCustomDelimiterPattern(String customDelimiterLine) {
+        // 커스텀 구분자 형식 체크
+        if (!isValidCustomDelimiter(customDelimiterLine)) {
+            throw new IllegalArgumentException("잘못된 구분자 형식입니다.");
```
문자열은 하드코딩하는 것보다, 상수 관리 클래스에서 관리하는 건 어떨까요?

m-a-king replied:
1. 여러 번 사용되지 않고
2. 의미가 드러나지 않는 매직 넘버나 스트링이 아닌 경우

이렇게 사용하는 게 더 좋았어요. 물론 확장성을 고려한다면 올바른 지적이 맞습니다

yeonjookang commented on src/test/java/calculator/InputTest.java:
통합테스트뿐만 아니라 유닛 테스트를 진행하면 오류가 어디서 나는지 빠르게 찾을 수 있을 것 같아요!

yeonjookang commented on src/test/java/calculator/InputTest.java:
```diff
@@ -0,0 +1,89 @@
+    }
+
+    @Test
+    @DisplayName("음수가 포함된 경우 예외가 발생해야 한다")
```
양수를 제외한 숫자는 경계값이 0이니까, 만약 예외 테스트를 한다면 0에 대해 테스트를 진행하는 게 좋을 것 같아요!

m-a-king replied:
테스트에 0이 빠진 것 짚어주셨군요. 감사합니다.
참고하겠습니다.

m-a-king replied:
감사합니다. 테스트에 0을 포함하지 않았네요!

yeonjookang commented on src/main/java/calculator/IOService.java:
```diff
@@ -0,0 +1,40 @@
+    public String getInput() {
+        try {
+            System.out.println("덧셈할 문자열을 입력해 주세요.");
+            String input = getSafeInput();
```
getSafeInput이랑 validateInput을 따로 구분하신 이유가 궁금해요! getSafeInput이라고 하면 유효성 검사가 포함되어있는 느낌이 들어서요!

m-a-king replied:
1. "공백이 문자열에 포함되면 안된다" -> 임의로 정한 요구 사항
2. 요구사항에서 주어진 빈 문자열 입력 처리

이렇게 구분했는데 메서드 명명이 애매했습니다. 참고하겠습니다!

yeonjookang reviewed (commented):
1주차 미션 수고하셨습니다 :)

holyPigeon commented on src/main/java/calculator/IOService.java:
```diff
@@ -0,0 +1,40 @@
+        }
+    }
+
+    public void printResult(BigDecimal result) {
+        System.out.println("결과 : " + result.toPlainString());
+    }
+}
```
매우 사소한 부분이지만, `printResult` 메서드는 그 역할에 따라 오직 "출력"만을 담당하는 것이 맞다고 생각합니다!

변환이 필요한 BigDecimal 타입으로 파라미터를 받는 것보다는, 파라미터를 String 타입으로 정의하고 `result.toPlainString()`을 주입받는 게 어떨까요?

m-a-king replied:
엄청 주의깊게 확인해주셨네요. 정말 감사합니다.
참고하겠습니다.

holyPigeon commented on src/main/java/calculator/NumberService.java:
```diff
@@ -0,0 +1,49 @@
+        return number;
+    }
+
+    private void validateNumberString(String numString) {
+        if (numString.isEmpty()) {
+            return;
+        }
+
+        if (numString.length() > 1 && numString.charAt(0) == '0' && numString.charAt(1) != '.') {
+            throw new IllegalArgumentException("숫자에 불필요한 0을 포함하지 않습니다.");
+        }
+
+        try {
+            new BigDecimal(numString); // BigDecimal로 변환해보고 유효성 검사
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("잘못된 숫자 형식입니다.");
+        }
+    }
```
여기서 개행으로 구분해주신 3개의 단락은 모두 하위 메서드로 분리해도 괜찮을 것 같다는 생각이 듭니다!

holyPigeon replied:
또한 가운데에 있는 if문은 조건절이 다소 길어서 해당 조건문을 `boolean hasUnnecessaryZero(String numString)` 등의 메서드로 빼도 괜찮다는 생각이 듭니다!

m-a-king replied:
올바른 지적이십니다. 참고하겠습니다.

holyPigeon commented on src/main/java/calculator/DelimiterService.java:
```diff
@@ -0,0 +1,53 @@
+        String[] lines = input.split("\\\\n");
+        String delimiterPattern = getDefaultDelimiterPattern();
+
+        if (lines.length == 2) {
+            delimiterPattern = getCustomDelimiterPattern(lines[0]);
+            return Arrays.asList(lines[1].split(delimiterPattern));
+        }
+
+        return Arrays.asList(lines[0].split(delimiterPattern));
```
line[0], line[1] 등이 의미하는 것이 무엇인지 파악하기 어려운 것 같습니다!

해당 값들을 의미를 변수로 나타내거나, 또는 부분 로직을 메서드로 묶어서 그 의미를 표현하는 것은 어떨까요?

m-a-king replied:
맞습니다. 그런 관점에서 확인하니 코드를 이해하기 상당히 번거롭네요. 감사합니다!

holyPigeon reviewed (commented):
요구사항 자체는 디테일하게 지키신 것 같아서 주로 코드 가독성 관련해서 말씀드렸습니다! 

1주차도 너무 고생하셨고, 2주차도 화이팅입니다 👍

m-a-king reviewed (commented):
좋은 코드 리뷰 감사드립니다. 많이 배웠습니다.

m-a-king committed:
chore(Coordinator): change IOservice 변수명

- 변수명 시작을 대문자에서 소문자로 변경하여 일반적인 자바 변수명 컨벤션을 따르도록 수정

m-a-king committed:
test(InputTest): add 0이 포함된 경우 예외 처리

- 요구사항에 따라서 양수만을 처리할 수 있으므로, 0을 포함한 테스트를 추가

m-a-king committed:
refactor(IOService): printResult 파라미터 타입 변경

- 출력 메서드가 책임에 집중할 수 있도록 변경

best11gh commented:
README 파일 작성을 진짜 꼼꼼히 하신 것 같네요. 이 구조는 SRP를 중점에 둔 코드인가요? 전 MVC 패턴만 써보아서 이 패턴만을 목표로 코드를 짰는데 전체적인 구조도 그렇고 Coordinator도 그렇고 신기하네요.

2주차도 화이팅입니다!

best11gh commented:
> README 파일 작성을 진짜 꼼꼼히 하신 것 같네요. 이 구조는 SRP를 중점에 둔 코드인가요? 전 MVC 패턴만 써보아서 이 패턴만을 목표로 코드를 짰는데 전체적인 구조도 그렇고 Coordinator도 그렇고 신기하네요.
> 
> 2주차도 화이팅입니다!

m-a-king commented:
@best11gh 

네, 말씀하신 대로 SRP를 중점에 둔 코드가 맞습니다. SRP를 엄격히 적용한다면 더 세분화할 수도 있겠지만, 적정 수준에서 클래스를 나누는 것이 더 실용적이라고 판단했습니다.

2주차 과제도 화이팅입니다!
